pulser/simulation/simulation.py

```diff
@@ -15,116 +15,302 @@
-
+        self._collapse_ops: list[qutip.Qobj] = []
+        self._times = self._adapt_to_sampling_rate(
+            np.arange(self._tot_duration, dtype=np.double)/1000)
+        self.evaluation_times = evaluation_times
+        self._config = config if config else SimConfig()
+        self._update_noise()
         self._extract_samples()
         self._build_basis_and_op_matrices()
         self._construct_hamiltonian()
+        if 'dephasing' in self.config.noise:
+            self._init_dephasing()
+        self.initial_state = 'all-ground'
+
+    @property
+    def config(self) -> SimConfig:
+        """Property getter for config."""
+        return self._config
+
+    def set_config(self, cfg: SimConfig) -> None:
+        """Sets current config to cfg and updates simulation parameters.
+
+        Args:
+            cfg (SimConfig): New configuration.
+        """
+        if not isinstance(cfg, SimConfig):
+            raise ValueError(f"Object {cfg} is not a valid `SimConfig`")
+        self._config = cfg
+        self._set_param_from_config()
+
+    def update_config(self, config: SimConfig) -> None:
+        """Updates this SimConfig object with parameters of another one.
+
+        Mostly useful when dealing with multiple noise types in different
+        configurations and wanting to merge these configurations together.
+        Updates simulation parameters only when dealing with noise types that
+        weren't available in the former SimConfig.
+
+        Args:
+            config (SimConfig): SimConfig to retrieve parameters from.
+        """
+        if not isinstance(config, SimConfig):
+            raise ValueError(f"Object {config} is not a valid `SimConfig`")
+
+        old_noise_set = set(self.config.noise)
+        new_noise_set = old_noise_set.union(config.noise)
+        diff_noise_set = new_noise_set - old_noise_set
+        param_dict: dict[str, Any] = asdict(self._config)
+        del param_dict["spam_dict"]
+        del param_dict["doppler_sigma"]
+        param_dict['noise'] = tuple(new_noise_set)
+        if 'SPAM' in diff_noise_set:
+            param_dict['eta'] = config.eta
+            param_dict['epsilon'] = config.epsilon
+            param_dict['epsilon_prime'] = config.epsilon_prime
+        if 'doppler' in diff_noise_set:
+            param_dict['temperature'] = config.temperature * 1.e6
+        if 'amplitude' in diff_noise_set:
+            param_dict['laser_waist'] = config.laser_waist
+        if 'dephasing' in diff_noise_set:
+            param_dict['dephasing_prob'] = config.dephasing_prob
+        self.set_config(SimConfig(**param_dict))
+
+    def show_config(self) -> None:
+        """Shows current configuration."""
+        print(self._config)
+
+    def reset_config(self) -> None:
+        """Resets configuration to default."""
+        self.set_config(SimConfig())
+        print("Configuration has been set to default.")
+
+    def _set_param_from_config(self) -> None:
```

Comment:
The comment I left below for `_prepare_bad_atoms` and `_prepare_doppler_detune` also applies here, since this is only called inside `set_config()`